src/utils/number_utils.ts

```diff
@@ -31,6 +36,18 @@ export type NumberUtils = {
  * @public
  */
 export const NumberUtils: NumberUtils = {
+  getSize(source: Uint8Array, offset: number): number {
```

Comment:
Is this really any better than:

```typescript
const size = NumberUtils.parseInt32LE(...);
if (size < 0) { .... }
```

Reply:
Yes, we can short-circuit this way avoiding unnecessary bit-wise operations.

Reply:
A small one but yes measurable